l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -221,6 +526,60 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         return balanceAfter - balanceBefore;
     }
 
+    function bridgehubDeposit(
+        uint256 _chainId,
+        address _prevMsgSender,
+        bytes calldata _data
+    ) external payable override onlyBridgehub returns (L2TransactionRequestTwoBridgesInner memory request) {
+        (address _l1Token, uint256 _amount, address _l2Receiver) = abi.decode(_data, (address, uint256, address));
+        require(msg.value == 0, "L1EB m.v > 0 for BH dep");
+        require(l2BridgeAddress[_chainId] != address(0), "L1EB b. n deployed");
+        require(bridgehub.baseToken(_chainId) != _l1Token, "L1EB base deposit");
+
+        require(_amount != 0, "6T"); // empty deposit amount
+        uint256 amount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _amount);
+        require(amount == _amount, "5T"); // The token has non-standard transfer logic
+
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId][_l1Token] += _amount;
+        }
+        bytes32 txDataHash = keccak256(abi.encodePacked(_prevMsgSender, _l1Token, _amount));
```

Comment:
Here you are using `abi.encodePacked`, while 
- https://github.com/matter-labs/era-contracts/blob/be9d49a851f4d498dc3d1eb08ed65d27b05e1c35/l1-contracts/contracts/bridge/L1ERC20Bridge.sol#L459

using `abi.encode`. You should to use one method for all `txDataHash`.

```suggestion
        bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, _l1Token, _amount));
```

Reply:
Yes, wanted to use abi.encode, thanks for the catch